src/helpers/BasicPointsAndBadges/PointsAndBadges.ts

```diff
@@ -6,16 +6,31 @@ import { SurveyCompletionActivity, awardSurveyCompletionActivityPoints } from ".
 import { CustomActivity, awardCustomActivityPoints } from "./CustomActivity";
 
 export async function awardPointsAndBadges(activities: BasicPointsForBadgesActivity[], state: BasicPointsForBadgesState, pointsPerBadge: number, participantInfo: ParticipantInfo): Promise<BasicPointsForBadgesState> {
-    const awardPointsPromises = activities.map((activity) => awardPointsForActivity(activity, state.activityStates[activity.key], participantInfo));
+
+    function getActivityState(activity: BasicPointsForBadgesActivity) : any{
+        if (Object.keys(state.activityStates).includes(activity.key)) {
+            return state.activityStates[activity.key];
+        }
+
+        return { pointsAwarded: 0 };
+    }
+
+    const inActiveActivityKeys = Object.keys(state.activityStates).filter((earnings) => activities.map(a => a.key).indexOf(earnings) === -1);
```

Comment:
Minor:  Couple rename suggestions.

- `inActiveActivityKeys` to `inactiveActivityKeys`
- 'earnings` to `key`